docs/gaps/teams-proposal/README.md

```diff
@@ -0,0 +1,310 @@
+include multiple teams. For example, given the above metadata, Alice can add a
+`QA-team` to the `authorize-ios-team` rule and set the rule threshold to 2,
+requiring both the `ios-team` and `QA-team` to sign off.
+
+The alternative to this approach is to either share keys between team members,
+or to implicitly define them through rules. The first complicates key
+management, and the latter does not allow for emergency updates, as an entity
+trusted for a namespace must always be trusted. The solution to the latter
+problem would be requiring users to use different keys for different contexts,
+which again complicates key management, and may create [out-of-sync principal
+definitions](/docs/gaps/5/README.md#out-of-sync-principal-definitions).
+
+## Backwards Compatibility
+
+Introducing teams and hat selection in gittuf must be backwards compatible, as
+the previous metadata definitions and verification flows of keys and principals
+remain the same. Repository administrators will still be able to use
+single-identity principals, and may incorporate teams in certain scenarios where
+groups of collaborators must be represented.
+
+## Security
+
+### Verification Workflow with Teams & Hats
+
+Supporting teams and hats introduces some additional complexity during
+verification. Verification must fail if a person is not wearing a valid hat. To
+enforce this, gittuf must validate the person's hat worn, either at the time of
+signing or retrospectively. If a hat specifier is omitted, gittuf assumes the
+default hat, which specifies the person-principal itself. The alternative
+default hat mechanism would be to trust the signature on all possible hats, but
+this may blur the intent of the signer and the context they were signing on.
+
+#### Counting a Team towards a Threshold
+
+As stated in [GAP-5](/docs/gaps/5/README.md): "The workflow must now track
+whether a key was used for another previously examined principal to avoid
+counting a single signature for two principals who both have the same key
+associated with them."
+
+For team thresholds, since gittuf does not support nested teams (a team
+consisted of another team as its member), the above workflow still applies. That
+is, counting signatures towards team thresholds must still track keys and make
+sure the same key cannot be used for more than 1 person on the team.
+
+For rule thresholds, the verification flow behaves a bit more leniently. This
+means that it does not explicitly track whether a person's signature was
+consumed for a previously examined team, allowing itself to count more than 1
+entity that use the same underlying key toward the rule threshold. Enabling this
+behavior can be useful in certain cases. For example, if Bob is responsible for
+both development and compliance on the branch, it may make sense to allow for
+Bob to sign on behalf of both teams. gittuf ultimately leaves this decision up
+to the delegating entity to decide.
+
```

Comment:
This question may be too general but, how does all of this relate to something like MFA where we may want to be able to have some MFA system also sign something / provide a temporal key?  So, if I want to bypass, what if it makes me use additional authentication mechanisms internally?  Can this be captured / used?